deployer/migrations/0004_change_service_id_to_ulid.sql

```diff
@@ -0,0 +1,107 @@
+-- Copy current service table. Keep the old table around because
+-- the rest of the tables are still having an FK on service_id.
+CREATE TABLE IF NOT EXISTS services_copy (
+    id TEXT PRIMARY KEY, -- Identifier of the service.
+    name TEXT UNIQUE     -- Name of the service.
+);
+
+INSERT INTO services_copy (id, name)
+SELECT
+  uuid_to_ulid(services.id),
+  services.name
+FROM services;
+
+-- Copy current deployments table without the FK service_id constraint.
+CREATE TABLE IF NOT EXISTS deployments_copy (
+    id TEXT PRIMARY KEY, -- Identifier of the deployment.
+    service_id TEXT,     -- Identifier of the service this deployment belongs to.
+    state TEXT,          -- Enum indicating the current state of the deployment.
+    last_update INTEGER, -- Unix epoch of the last status update
+    address TEXT,        -- Address a running deployment is active on
+    is_next BOOLEAN,     -- Whether the deployment is for a shuttle-next runtime
+    git_commit_id TEXT,  -- Deployment git commit id
+    git_commit_msg TEXT, -- Deployment last git commit msg
+    git_branch TEXT,     -- Deployment git branch
+    git_dirty BOOLEAN    -- Deployment git state is dirty
+);
+
+INSERT INTO deployments_copy (id, service_id, state, last_update, address, is_next)
+SELECT
+  deployments.id,
+  uuid_to_ulid(deployments.service_id),
+  deployments.state,
+  deployments.last_update,
+  deployments.address,
+  deployments.is_next,
+  deployments.git_commit_id,
+  deployments.git_commit_msg,
+  deployments.git_branch,
+  deployments.git_dirty
+FROM deployments;
+
+-- Copy current resource table without the FK service_id constraint.
+CREATE TABLE IF NOT EXISTS resources_copy (
+    service_id TEXT,   -- Identifier of the service this resource belongs to.
+    type TEXT,         -- Type of resource this is.
+    data TEXT,         -- Data about this resource.
+    config TEXT,       -- Resource configuration.
+    PRIMARY KEY (service_id, type),
+);
+INSERT INTO resources_copy (service_id, type, data, config)
+SELECT
+  uuid_to_ulid(resources.service_id),
```

Comment:
Is this deterministic?

Reply:
Also, shouldn't the ulid0 plugin be loaded in `persistence/mod.rs` for this to work?

Reply:
> Is this deterministic?

I think so: https://github.com/dylanhart/ulid-rs/blob/master/src/uuid.rs#L8.

>  Also, shouldn't the ulid0 plugin be loaded in persistence/mod.rs for this to work?

Thanks, good catch. Fixed it.

Reply:
We haven't used `uuid_to_ulid` because the uuid misses the time information, which makes the associated ulid unpredictable in terms of time information. Used `uuid_with_datetime` instead with '2023-06-01` as time info for the services ulids that were created before releasing this PR.